tests/Aspire.Azure.AI.OpenAI.Tests/AspireAzureAIOpenAIExtensionsTests.cs

```diff
@@ -0,0 +1,98 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Azure.AI.OpenAI;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+using Xunit;
+
+namespace Aspire.Azure.AI.OpenAI.Tests;
+
+public class AspireAzureAIOpenAIExtensionsTests
+{
+    private const string ConnectionString = "Endpoint=https://aspireopenaitests.openai.azure.com/;Key=fake";
+
+    [Theory]
+    [InlineData(true)]
+    [InlineData(false)]
+    public void ReadsFromConnectionStringsCorrectly(bool useKeyed)
+    {
+        var builder = Host.CreateEmptyApplicationBuilder(null);
+        builder.Configuration.AddInMemoryCollection([
+            new KeyValuePair<string, string?>("ConnectionStrings:openai", ConnectionString)
+        ]);
+
+        if (useKeyed)
+        {
+            builder.AddKeyedAzureOpenAI("openai");
+        }
+        else
+        {
+            builder.AddAzureOpenAI("openai");
+        }
+
+        var host = builder.Build();
+        var client = useKeyed ?
+            host.Services.GetRequiredKeyedService<OpenAIClient>("openai") :
+            host.Services.GetRequiredService<OpenAIClient>();
+
+        Assert.NotNull(client);
+    }
+
+    [Theory]
+    [InlineData(true)]
+    [InlineData(false)]
+    public void ConnectionStringCanBeSetInCode(bool useKeyed)
+    {
+        var uri = new Uri("https://aspireopenaitests.openai.azure.com/");
+        var key = "fake";
+        var builder = Host.CreateEmptyApplicationBuilder(null);
+
+        if (useKeyed)
+        {
+            builder.AddKeyedAzureOpenAI("openai", settings => { settings.Endpoint = uri; settings.Key = key; });
+        }
+        else
+        {
+            builder.AddAzureOpenAI("openai", settings => { settings.Endpoint = uri; settings.Key = key; });
+        }
+
+        var host = builder.Build();
+        var client = useKeyed ?
+            host.Services.GetRequiredKeyedService<OpenAIClient>("openai") :
+            host.Services.GetRequiredService<OpenAIClient>();
+
+        Assert.NotNull(client);
+    }
+
+    [Theory]
+    [InlineData(true)]
+    [InlineData(false)]
+    public void ConnectionNameWinsOverConfigSection(bool useKeyed)
+    {
+        var builder = Host.CreateEmptyApplicationBuilder(null);
+
+        var key = useKeyed ? "openai" : null;
+        builder.Configuration.AddInMemoryCollection([
+            new KeyValuePair<string, string?>(ConformanceTests.CreateConfigKey("Aspire:Azure:AI.OpenAI", key, "ConnectionString"), "unused"),
```

Comment:
AzureOpenAISettings doesn't support a `ConnectionString` property, so this is kind of pointless test.

Reply:
It implements `ParseConnectionString` which is invoked by this test, even though it is not setting `ConnectionString` property specifically (but Endpoint;Key pair)

Reply:
Ok, got it, this line is wrong but the test is still valid.

Reply:
It's not really valid. The test name is `ConnectionNameWinsOverConfigSection`. Now that you've removed the config section, what is the connection name winning over?

Reply:
Right, and this "new" case is now already tested too (reading from ConnectionStrings section), removed.